core/src/main/java/de/javakaffee/web/msm/LockingStrategy.java

```diff
@@ -388,50 +378,59 @@ protected void onAfterDeleteFromMemcached( @Nonnull final String sessionId ) {
             return false;
         }
+
+        if ( touchResult ) {
+            _stats.nonStickySessionsPingFailed();
+            _log.warn( "The session " + sessionId
+                    + " should be touched in memcached, but it does not exist therein." );
+            return false;
+        }
+        else {
+            _log.debug( "The session was ping'ed successfully." );
+            return true;
+        }
     }
 
     private void pingSession( @Nonnull final MemcachedBackupSession session,
-            @Nonnull final BackupSessionService backupSessionService ) throws InterruptedException {
-        final Future<Boolean> touchResult = _memcached.add( _storageKeyFormat.format(session.getIdInternal()), 5, 1 );
+            @Nonnull final BackupSessionService backupSessionService ) {
+        boolean touchResult;
         try {
-            if ( touchResult.get() ) {
-                _stats.nonStickySessionsPingFailed();
-                _log.warn( "The session " + session.getIdInternal()
-                        + " should be touched in memcached, but it does not exist"
-                        + " therein. Will store in memcached again." );
-                updateSession( session, backupSessionService );
-            }
-            else
-                _log.debug( "The session was ping'ed successfully." );
-        } catch ( final ExecutionException e ) {
+            touchResult = _memcached.add( _storageKeyFormat.format(session.getIdInternal()), 5, BYTE_1 );
+        } catch ( final IOException e ) {
             _log.warn( "An exception occurred when trying to ping session " + session.getIdInternal(), e );
+            return;
         }
+        
+        if ( touchResult ) {
+            _stats.nonStickySessionsPingFailed();
+            _log.warn( "The session " + session.getIdInternal()
+                    + " should be touched in memcached, but it does not exist"
+                    + " therein. Will store in memcached again." );
+            updateSession( session, backupSessionService );
+        }
+        else
+            _log.debug( "The session was ping'ed successfully." );
     }
 
     private void updateSession( @Nonnull final MemcachedBackupSession session,
-            @Nonnull final BackupSessionService backupSessionService ) throws InterruptedException {
+            @Nonnull final BackupSessionService backupSessionService ) {
         final Future<BackupResult> result = backupSessionService.backupSession( session, true );
         try {
             if ( result.get().getStatus() != BackupResultStatus.SUCCESS ) {
                 _log.warn( "Update for session (after unsuccessful ping) did not return SUCCESS, but " + result.get() );
             }
         } catch ( final ExecutionException e ) {
             _log.warn( "An exception occurred when trying to update session " + session.getIdInternal(), e );
+        } catch ( final InterruptedException e ) {
```

Comment:
Why do you catch this InterruptedException? If there's a good reason to catch it, at least the interrupted status should be restored.